my-app/src/ListIterator.js

```diff
@@ -0,0 +1,63 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+class ListIterator extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      listItems: this.props.listItems
+    };
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (prevState.listItems !== nextProps.listItems) {
+      let sortedArray = nextProps.listItems.sort();
```

Comment:
do sorting before render or create an array with the sorted element so every time u do not hv to sort the array